src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/ShardedHttpExecutionServiceCluster.scala

```diff
@@ -0,0 +1,115 @@
+
+  // following are called on a workflow that has already been submitted.
+  // therefore, we want to use the cromwell instance that has been persisted
+  // onto that workflow.
+  def status(wfe: WorkflowExecution, userInfo: UserInfo): Future[ExecutionServiceStatus] =
+    getMember(wfe).flatMap {memb => memb.status(wfe.id, userInfo)}
+
+  def callLevelMetadata(wfe: WorkflowExecution, userInfo: UserInfo): Future[ExecutionMetadata] =
+    getMember(wfe).flatMap {memb => memb.callLevelMetadata(wfe.id, userInfo)}
+
+  def outputs(wfe: WorkflowExecution, userInfo: UserInfo): Future[ExecutionServiceOutputs] =
+    getMember(wfe).flatMap {memb => memb.outputs(wfe.id, userInfo)}
+
+  def logs(wfe: WorkflowExecution, userInfo: UserInfo): Future[ExecutionServiceLogs] =
+    getMember(wfe).flatMap {memb => memb.logs(wfe.id, userInfo)}
+
+  def abort(wfe: WorkflowExecution, userInfo: UserInfo): Future[Try[ExecutionServiceStatus]] =
+    getMember(wfe).flatMap {memb => memb.abort(wfe.id, userInfo)}
+
+
+  // ====================
+  // facade-to-cluster entry points
+  // ====================
+  // for an already-submitted workflow, get the instance to which it was submitted
+  private def getMember(wfe: WorkflowExecution):Future[ExecutionServiceDAO] = wfe.executionServiceId match {
+    case Some(execId) => Future.successful(getMemberById(ExecutionServiceId(execId)))
+    case None => {
+      dataSource.inTransaction { dataAccess =>
+        dataAccess.workflowQuery.getExecutionServiceKey(wfe.id) map {execIdOption =>
+          execIdOption match {
+            case Some(execId) => getMemberById(ExecutionServiceId(execId))
+            case None => getDefaultMember //throw new RawlsException("can only process Workflow objects with an execution service key")
+          }
+        }
+      }
+    }
+  }
+
+  // for unsubmitted workflows, get the best instance to which we should submit
+  // we expect that all workflowRecs passed to this method will be a single batch;
+  // we return one target cromwell instance for the entire batch.
+  private def nextAvailableMember(workflowRecs: Seq[WorkflowRecord]): ClusterMember = {
+    // inspect the first workflow in the batch, retrieve a long value of its last-updated timestamp
+    // (could also use workflowId or some other number)
+    val shardingSeed = workflowRecs.head.statusLastChangedDate.getTime
+    val shardIndex = targetIndex(shardingSeed, members.size)
+
+    memberArray(shardIndex)
+  }
+
+
+
+  // ====================
+  // clustering methods
+  // ====================
+  private def getMemberById(key: ExecutionServiceId): ExecutionServiceDAO = members.get(key).orElse(throw new RawlsException(s"member with key $key does not exist")).get
+
+  private def getDefaultEntry:(ExecutionServiceId, ExecutionServiceDAO) = defaultEntry
+  def getDefaultMember: ExecutionServiceDAO = getDefaultEntry._2
+
+  def targetIndex(seed: Long, numTargets: Int):Int = Math.ceil( (seed % 100) / (100 / numTargets) ).toInt
```

Comment:
HAHAHA I suppose that would be simpler, wouldn't it? I really wanted to prove out a code path that inspects the inbound workflow(s), and routes based on some criteria of that workflow. By the time I ended up here ... I never looked back to re-simplify.